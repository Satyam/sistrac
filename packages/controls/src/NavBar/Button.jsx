import React from 'react';
import PropTypes from 'prop-types';
import { Link, Route } from 'react-router-dom';
import classNames from 'classnames';

const Button = ({ children, href, disabled, external, onClick, className }) => {
  if (disabled) {
    return (
      <li className={classNames('nav-item', className)}>
        <a className="nav-link disabled" aria-disabled>
          {children}
        </a>
      </li>
    );
  }
  if (external || onClick) {
    return (
      <li className={classNames('nav-item', className)}>
        <a className="nav-link" href={href} onClick={onClick}>
          {children}
        </a>
      </li>
    );
  }
  return (
    <Route path={href}>
      {({ match }) => (
        <li className={classNames('nav-item', className, { active: match })}>
          <Link className="nav-link" to={href}>
            {children}
          </Link>
        </li>
      )}
    </Route>
  );
};

Button.propTypes = {
  children: PropTypes.node,
  href: PropTypes.string,
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
  external: PropTypes.bool,
  className: PropTypes.string,
};

export default Button;
