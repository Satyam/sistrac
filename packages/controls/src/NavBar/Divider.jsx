import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const Divider = ({ className, ...props }) => (
  <div className={classNames('dropdown-divider', className)} {...props} />
);

Divider.propTypes = {
  className: PropTypes.string,
};

export default Divider;
