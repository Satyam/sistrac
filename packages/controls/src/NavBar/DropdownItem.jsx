import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import classNames from 'classnames';

const DropdownItem = ({ children, href, disabled, className, external }) =>
  disabled ? (
    <a
      className={classNames('dropdown-item disabled', className)}
      style={{ opacity: 0.3 }}
    >
      {children}
    </a>
  ) : external ? (
    <a className={classNames('dropdown-item', className)} href={href}>
      {children}
    </a>
  ) : (
    <Link className={classNames('dropdown-item', className)} to={href}>
      {children}
    </Link>
  );

DropdownItem.propTypes = {
  children: PropTypes.node,
  href: PropTypes.string,
  disabled: PropTypes.bool,
  external: PropTypes.bool,
  className: PropTypes.string,
};

export default DropdownItem;
