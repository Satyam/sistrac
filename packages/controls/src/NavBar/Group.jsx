import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const Group = ({ children, right }) => (
  <ul
    className={classNames('navbar-nav', {
      'd-flex justify-content-end ml-auto': right,
    })}
  >
    {children}
  </ul>
);

Group.propTypes = {
  children: PropTypes.node,
  right: PropTypes.bool,
};

export default Group;
