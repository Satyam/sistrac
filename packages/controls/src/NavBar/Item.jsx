import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import classNames from 'classnames';

const Item = ({ children, href, disabled, className, external, onClick }) => {
  if (disabled) {
    return (
      <a
        className={classNames('dropdown-item disabled', className)}
        style={{ opacity: 0.3 }}
      >
        {children}
      </a>
    );
  }
  if (external) {
    return (
      <a
        className={classNames('dropdown-item', className)}
        href={href}
        onClick={onClick}
      >
        {children}
      </a>
    );
  }
  return (
    <Link className={classNames('dropdown-item', className)} to={href}>
      {children}
    </Link>
  );
};

Item.propTypes = {
  children: PropTypes.node,
  href: PropTypes.string,
  disabled: PropTypes.bool,
  external: PropTypes.bool,
  className: PropTypes.string,
  onClick: PropTypes.func,
};

export default Item;
