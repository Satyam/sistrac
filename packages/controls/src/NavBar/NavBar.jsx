import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import classNames from 'classnames';

class NavBar extends Component {
  state = { open: true };
  toggle = ev => {
    ev.preventDefault();
    this.setState({ open: !this.state.open });
  };
  dismiss = ev => {
    if (ev.target.href) this.setState({ open: false });
  };
  render() {
    const {
      children,
      brand,
      href,
      position,
      dark,
      background,
      breakpoint,
      external,
      onClick,
    } = this.props;
    const { open } = this.state;
    return (
      <nav
        className={classNames(
          'navbar',
          {
            'fixed-top': position === 'top',
            'fixed-bottom': position === 'bottom',
            'sticky-top': position === 'sticky-top',
            'navbar-light': !dark,
            'navbar-dark': dark,
          },
          background && `bg-${background}`,
          breakpoint && `navbar-expand-${breakpoint}`,
        )}
      >
        {href ? (
          external ? (
            <a href={href} className="navbar-brand">
              {brand}
            </a>
          ) : (
            <Link to={href} className="navbar-brand">
              {brand}
            </Link>
          )
        ) : (
          <div className="navbar-brand" onClick={onClick}>
            {brand}
          </div>
        )}
        <button
          className={classNames('navbar-toggler', { collapsed: open })}
          type="button"
          onClick={this.toggle}
          aria-expanded={!open}
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon" />
        </button>
        <div
          className={classNames('collapse', 'navbar-collapse', {
            show: open,
          })}
          onClick={this.dismiss}
        >
          {children}
        </div>
      </nav>
    );
  }
}

NavBar.propTypes = {
  children: PropTypes.node,
  brand: PropTypes.node,
  href: PropTypes.string,
  external: PropTypes.bool,
  position: PropTypes.oneOf(['top', 'bottom', 'sticky-top']),
  dark: PropTypes.bool,
  background: PropTypes.oneOf([
    'primary',
    'secondary',
    'info',
    'warning',
    'danger',
    'light',
    'white',
    'dark',
  ]),
  breakpoint: PropTypes.oneOf(['sm', 'md', 'lg', 'xl']),
  onClick: PropTypes.func,
};

export default NavBar;
