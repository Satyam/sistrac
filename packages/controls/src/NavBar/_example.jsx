import React from 'react';
import NavBar from './';
import FaBeer from 'react-icons/lib/fa/beer';
import Search from 'react-icons/lib/go/search';

const handlSearch = value => {
  console.log('search for', value);
};
const example = () => (
  <div style={{ paddingTop: '4em' }}>
    <p>first item</p>
    <p>lots of nonsense</p>
    <p>lots of nonsense</p>
    <p>lots of nonsense</p>
    <p>lots of nonsense</p>
    <p>end of top</p>
    <NavBar
      brand="algo"
      href="pepe"
      position="top"
      breakpoint="md"
      background="light"
      external
    >
      <NavBar.Group>
        <NavBar.Button href="/jose">Menu 1</NavBar.Button>
        <NavBar.Button href="/jose" disabled>
          Menu 2
        </NavBar.Button>
        <NavBar.Menu
          label={
            <span>
              <FaBeer /> drpdwn
            </span>
          }
          disabled
        >
          <NavBar.Item href="pepe1">pepe1</NavBar.Item>
          <NavBar.Item href="pepe2">pepe2</NavBar.Item>
          <NavBar.Item href="pepe3">pepe3</NavBar.Item>
        </NavBar.Menu>
      </NavBar.Group>
      <NavBar.Group right>
        <NavBar.Search onSearch={handlSearch} color="success" outline>
          <Search />
        </NavBar.Search>
        <NavBar.Menu label="drpdwnR">
          <NavBar.Item href="pepe1">pepe1</NavBar.Item>
          <NavBar.Item href="pepe2" disabled>
            pepe2
          </NavBar.Item>
          <NavBar.Divider />
          <NavBar.Item href="pepe3">pepe3</NavBar.Item>
        </NavBar.Menu>
      </NavBar.Group>
    </NavBar>
    <p>start of bottom</p>
    <p>lots of nonsense</p>
    <p>lots of nonsense</p>
    <p>lots of nonsense</p>
    <p>lots of nonsense</p>
    <p>lots of nonsense</p>
    <p>last item</p>
  </div>
);

export default example;
