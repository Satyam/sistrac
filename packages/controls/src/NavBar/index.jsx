import NavBar from './NavBar';
import Group from './Group';
import Button from './Button';
import Menu from './Menu';
import Item from './Item';
import Divider from './Divider';
import Search from './Search';

NavBar.Group = Group;
NavBar.Button = Button;
NavBar.Menu = Menu;
NavBar.Item = Item;
NavBar.Divider = Divider;
NavBar.Search = Search;

export default NavBar;
export { NavBar, Group, Button, Menu, Item, Divider, Search };
