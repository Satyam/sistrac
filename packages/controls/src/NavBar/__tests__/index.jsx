import React from 'react';
import { mount } from 'enzyme';
import NavBar from '../';

describe('NavBar with Brand', () => {
  it('should render within one another', () => {
    const wrapper = mount(<NavBar brand="Titulo" />);
    expect(wrapper).toMatchSnapshot();
  });
});
