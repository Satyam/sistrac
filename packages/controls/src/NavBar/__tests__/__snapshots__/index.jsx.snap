// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar with Brand should render within one another 1`] = `
<NavBar
  brand="Titulo"
>
  <nav
    className="navbar navbar-light"
  >
    <div
      className="navbar-brand"
    >
      Titulo
    </div>
    <button
      aria-expanded={false}
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse show"
      onClick={[Function]}
    />
  </nav>
</NavBar>
`;
